bindings/python/requirements.txt

```diff
@@ -1,14 +1,12 @@
-maturin==1.0.1
-virtualenv==20.23.1
-ruff==0.0.275
+# this requirements.txt is for development deps only
+maturin[patchelf]==1.4.0
+patchelf
```

Comment:
we shouldn't need this. The `patchelf` dependency breaks my install on mac.

Reply:
@tychoish the `patchelf` dependency is still breaking things locally. We need to remove this.

Reply:
done